### ISSUE TITLE
0.6.14 Interpolation Fails Silently when Interpolation Syntax Contains Invalid Value

### ISSUE THREAD

**Issue by eedwardsdisco** [hashicorp/terraform#5957]
Edit: See my second post below, I re-phrased this issue due to the below behavior.

This is the third time I'm now getting instances of terraform just failing to run interpolation on a field.

It's getting VERY frustrating.

```
resource "aws_autoscaling_group" "autoscaling_group" {
    name = "${var.stagedocs_service_prefix_name}_service_${var.stagedocs_service_environment_name}"
    min_size = "${var.autoscaling_group_min_size}"
    max_size = "${var.autoscaling_group_max_size}"
    desired_capacity = "${var.autoscaling_group_desired_capacity}"

    vpc_zone_identifier = ["${compact(split(",", var.autoscaling_group_subnet_ids))}"]

    tag {
        key = "Name"
        value = "${var.stagedocs_service_prefix_name}_service_${var.stagedocs_service_environment_name}-v${stagedocs_service_version}-auto"
        propagate_at_launch = true
    }

    tag {
        key = "Category"
        value = "${var.autoscaling_group_tag_category}"
        propagate_at_launch = true
    }

    tag {
        key = "Team"
        value = "${var.autoscaling_group_tag_team}"
        propagate_at_launch = true
    }

    tag {
        key = "DataDog"
        value = "Monitored"
        propagate_at_launch = true
    }

    launch_configuration = "${aws_launch_configuration.launch_configuration.name}"

    health_check_type = "EC2"

    health_check_grace_period = "${var.autoscaling_group_health_check_grace_period}"

    force_delete = "${var.autoscaling_group_force_delete}"

    enabled_metrics = ["${compact(split(",", var.autoscaling_group_enabled_metrics))}"]
}
```

when I plan\apply...

```
aws_autoscaling_group.autoscaling_group: Creating...
  availability_zones.#:                "" => "<computed>"
  default_cooldown:                    "" => "<computed>"
  desired_capacity:                    "" => "1"
  force_delete:                        "" => "0"
  health_check_grace_period:           "" => "0"
  health_check_type:                   "" => "EC2"
  launch_configuration:                "" => "Stagedocs_service_edwards_1.0.243_a5xaxwnmirgsrdxenva7g6uce4"
  max_size:                            "" => "1"
  metrics_granularity:                 "" => "1Minute"
  min_size:                            "" => "1"
  name:                                "" => "Stagedocs_service_edwards"
  tag.#:                               "" => "4"
  tag.1667719954.key:                  "" => "Category"
  tag.1667719954.propagate_at_launch:  "" => "1"
  tag.1667719954.value:                "" => "Dev"
  tag.1730917790.key:                  "" => "Team"
  tag.1730917790.propagate_at_launch:  "" => "1"
  tag.1730917790.value:                "" => "DevOps"
  tag.697684838.key:                   "" => "DataDog"
  tag.697684838.propagate_at_launch:   "" => "1"
  tag.697684838.value:                 "" => "Monitored"
  tag.~1530607177.key:                 "" => "Name"
  tag.~1530607177.propagate_at_launch: "" => "1"
  tag.~1530607177.value:               "" => "${var.stagedocs_service_prefix_name}_service_${var.stagedocs_service_environment_name}-v${stagedocs_service_version}-auto"
  vpc_zone_identifier.#:               "" => "2"
  vpc_zone_identifier.1314723983:      "" => "<redacted>"
  vpc_zone_identifier.456478033:       "" => "<redacted>"
  wait_for_capacity_timeout:           "" => "10m"
aws_autoscaling_group.autoscaling_group: Creation complete
```

It's not interpolating my tag value! What the heck?! :(

**Comment by sl1pm4t** [comment]
What happens if you fix your interpolation variable reference:
`${stagedocs_service_version}`  --> `${var.stagedocs_service_version}`

**Comment by eedwardsdisco** [comment]
Yup! That fixed it! Thank you...

Ugh, this is what happens when you go back to coding right after lunch :P

So, does anyone know if that's expected behavior, that using the interpolation syntax `${}` with an invalid string variable inside causes the entire field to be silently unparsed \ fail parsing?

**Comment by catsby** [comment]
Hey all – thanks for finding the issue here :smile: 
Can you tell me, is there any extra information if you were to plan with `TF_LOG=1` that would reveal the issue there? Any warning message regarding the interpolation?

**Comment by eedwardsdisco** [comment]
Hi @catsby 

Nothing in the TRACE log that I saw (holy cow that's a lot of log!). I grepped through it for the variable name itself (as it was written, without the `var.` first) and looked for any warnings. Grepped for permutations of 'error', 'warning', 'interpolation', and visually scanned the file in sublime, but nothing I saw that looked like a warning or error message for interpolation.

**Comment by clstokes** [comment]
I ran into this as well in Terraform v0.7.10.

Here's a quick repro case and trace log (with keys removed): https://gist.github.com/clstokes/9cfa3029aea30ba7fa6de2e343f9cb59

**Comment by mitchellh** [comment]
This is fixed in TF 0.8!
